src/sentry/utils/sdk.py

```diff
@@ -124,6 +130,18 @@ def on_full_queue(*args, **kwargs):
     transport._worker.on_full_queue = on_full_queue
 
 
+def traces_sampler(sampling_context):
+    if "wsgi_environ" in sampling_context:
+        match = resolve(sampling_context["wsgi_environ"].get("PATH_INFO"))
+        if match and match.url_name in SAMPLED_URL_NAMES:
+            return 1.0
+
+    if sampling_context["parent_sampled"] is not None:
+        return sampling_context["parent_sampled"]
+
+    return 0.0
```

Comment:
I presume your ordering here is intentional, but it does mean that:

a) For situations in which this endpoint is hit from our front end, the front end sampling decision will be overridden, which will lead to incomplete traces when the front end said no but this forces a yes, and you get this transaction but not its parent. IDK if that's something that's important to you or not.

b) The work of doing the resolving and the url-name lookup is happening for every request, even though most of them don't need it. This might not be a big deal, if both of the lookups are constant time (IDK, but I assume the resolver uses a dictionary under the hood, and see my note above re: your check), but I still think you could get around it, even with the precedence order you've set up. It seems to me that given that you're always saying yes, the only transactions you might possibly need to change are those which come in with a no. So you might check that first, return the parent decision right away if it's yes, and only do the rest for the nos. What do you think?

Reply:
Changing the order here makes a lot of sense :+1:, and still maintains the behaviour I'm expecting. We want to keep the frontend decision if its already made one. And these are endpoints that we want to track when they're used outside of the frontend.